core/src/main/scala/stainless/extraction/imperative/Trees.scala

```diff
@@ -33,6 +33,22 @@ trait Trees extends oo.Trees with Definitions { self =>
       }
   }
 
+  /* Cell Operations */
+
+    /** Swap values from two (not necessarily distinct) cells */
+  sealed case class CellSwap(cell1: Expr, cell2: Expr) extends Expr with CachingTyped {
+    override protected def computeType(using Symbols): Type =
+      
+      (cell1.getType, cell2.getType) match {
+        case (ClassType(id1, tps1), ClassType(id2, tps2)) if id1.name == "Cell" && id2.name == "Cell" && tps1 == tps2=> {
+          // throw new Exception(s"CellSwap: ${id1} + $tps1, ${cell2.getType}") // TODO: remove
+          UnitType() // TODO
+        }
+        case _ =>
+          Untyped
```

Comment:
@Mario-bucev if I do not create an `Expr` for the cells, is it a proper way of doing things here?

Reply:
With `symbols.lookup.get[ClassDef]("stainless.lang.Cell")` you should get the `ClassDef` and in particular its `id`. Then you can check that `id1` and `id2` are the same as the `id` for `"stainless.lang.Cell"`.

Reply:
Perfect thank you!! I pushed the new version :)